manifest/v1alpha/report/validation.go

```diff
@@ -0,0 +1,106 @@
+package report
+
+import (
+	"github.com/pkg/errors"
+
+	validationV1Alpha "github.com/nobl9/nobl9-go/internal/manifest/v1alpha"
+	"github.com/nobl9/nobl9-go/internal/validation"
+	"github.com/nobl9/nobl9-go/manifest"
+	"github.com/nobl9/nobl9-go/manifest/v1alpha"
+)
+
+const (
+	expectedNumberOfReportTypes = 1
+)
+
+func validate(r Report) *v1alpha.ObjectError {
+	return v1alpha.ValidateObject(validator, r, manifest.KindReport)
+}
+
+var validator = validation.New(
+	validationV1Alpha.FieldRuleAPIVersion(func(r Report) manifest.Version { return r.APIVersion }),
+	validationV1Alpha.FieldRuleKind(func(r Report) manifest.Kind { return r.Kind }, manifest.KindReport),
+	validation.For(func(r Report) Metadata { return r.Metadata }).
+		Include(metadataValidation),
+	validation.For(func(r Report) Spec { return r.Spec }).
+		WithName("spec").
+		Include(specValidation),
+)
+
+var metadataValidation = validation.New[Metadata](
+	validationV1Alpha.FieldRuleMetadataName(func(m Metadata) string { return m.Name }),
+	validationV1Alpha.FieldRuleMetadataDisplayName(func(m Metadata) string { return m.DisplayName }),
+)
+
+var specValidation = validation.New[Spec](
+	validation.For(validation.GetSelf[Spec]()).
+		Rules(validation.NewSingleRule(func(s Spec) error {
+			reportTypeCounter := 0
+			if s.SystemHealthReview != nil {
+				reportTypeCounter++
+			}
+			if s.SLOHistory != nil {
+				reportTypeCounter++
+			}
+			if s.ErrorBudgetStatus != nil {
+				reportTypeCounter++
+			}
+			if reportTypeCounter != expectedNumberOfReportTypes {
+				return errors.New("exactly one report type configuration is required")
+			}
+			return nil
+		})),
+	validation.ForPointer(func(s Spec) *Filters { return s.Filters }).
+		WithName("filters").
+		Required().
+		Include(filtersValidation),
+	validation.ForPointer(func(s Spec) *SLOHistoryConfig { return s.SLOHistory }).
+		WithName("sloHistory").
+		Include(sloHistoryValidation),
+	validation.ForPointer(func(s Spec) *SystemHealthReviewConfig { return s.SystemHealthReview }).
+		WithName("systemHealthReview").
+		Include(systemHealthReviewValidation),
+)
+
+var filtersValidation = validation.New[Filters](
```

Comment:
We talked about introducing trivial project name <-> service/SLO project validation here.

Reply:
Validation for nested project/service/SLO implemented elsewhere.